src/data-sources/components/HttpAuthSettingsInput.tsx

```diff
@@ -0,0 +1,84 @@
+import { SelectControl, TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { ChangeEvent } from 'react';
+
+import PasswordInputControl from '@/data-sources/components/PasswordInputControl';
+import {
+	HTTP_SOURCE_AUTH_TYPE_SELECT_OPTIONS,
+	HTTP_SOURCE_ADD_TO_SELECT_OPTIONS,
+} from '@/data-sources/constants';
+import { HttpAuthFormState } from '@/data-sources/types';
+
+interface HttpAuthSettingsInputProps {
+	auth: HttpAuthFormState;
+	onChange: ( id: string, value: unknown ) => void;
+}
+
+export const HttpAuthSettingsInput: React.FC< HttpAuthSettingsInputProps > = ( {
+	auth,
+	onChange,
+} ) => {
+	const onSelectChange = (
+		value: string,
+		extra?: { event?: ChangeEvent< HTMLSelectElement > }
+	) => {
+		if ( extra?.event ) {
+			const { id } = extra.event.target;
+			onChange( id, value );
+		}
+	};
+
+	return (
+		<>
+			<div className="form-group">
+				<SelectControl
+					id="authType"
+					label={ __( 'Authentication Type', 'remote-data-blocks' ) }
+					value={ auth.authType }
+					onChange={ onSelectChange }
+					options={ HTTP_SOURCE_AUTH_TYPE_SELECT_OPTIONS }
+				/>
+			</div>
+
+			<div className="form-group">
+				<PasswordInputControl
+					id="authValue"
+					label={ __( 'Authentication Value', 'remote-data-blocks' ) }
+					value={ auth.authValue }
+					onChange={ value => onChange( 'authValue', value ) }
+					__next40pxDefaultSize
+					help={ __(
+						'The authentication value to use for the HTTP endpoint. When using Basic Auth, this is "username:password" string.',
+						'remote-data-blocks'
+					) }
+				/>
+			</div>
+			{ auth.authType === 'api-key' && (
+				<>
+					<div className="form-group">
+						<TextControl
+							id="authKey"
```

Comment:
It's useful to have `help`text on this field as well.

Reply:
8e995ae